### ISSUE TITLE
StackOverflowError ControllerHostedRouter.getTransactionIndexer

### ISSUE THREAD

**Issue by ghost** [bluelinelabs/Conductor#455]
Haven't personally been able to replicate yet but seeing a large number of crashes on our production build. We are using version 2.1.4.

`Caused by java.lang.StackOverflowError: stack size 8MB
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:239)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:239)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:239)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:239)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:239)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:239)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:239)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:239)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:239)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:239)
       at com.bluelinelabs.conductor.Router.getActivity(Router.java:735)
       at com.bluelinelabs.conductor.Router.getActivity(Router.java:726)
       at com.bluelinelabs.conductor.Router.pushController(Router.java:179)
       at com.pl.goodwood.home.HomeMasterController.pushController(HomeMasterController.kt:177)
       at com.pl.goodwood.home.HomeMasterController.navigateToFilteredList(HomeMasterController.kt:130)
       at com.pl.goodwood.home.HomeMasterPresenter$load$5.accept(HomeMasterPresenter.kt:40)
       at com.pl.goodwood.home.HomeMasterPresenter$load$5.accept(HomeMasterPresenter.kt:14)
       at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
       at io.reactivex.internal.operators.observable.ObservableSwitchMap$SwitchMapObserver.drain(ObservableSwitchMap.java:276)
       at io.reactivex.internal.operators.observable.ObservableSwitchMap$SwitchMapInnerObserver.onNext(ObservableSwitchMap.java:328)
       at io.reactivex.internal.operators.observable.ObservableDoOnEach$DoOnEachObserver.onNext(ObservableDoOnEach.java:103)
       at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:200)
       at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:252)
       at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:109)
       at android.os.Handler.handleCallback(Handler.java:790)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6753)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:482)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)`

**Comment by ar-g** [comment]
The conductor is the crash number 1 in our crashlytics, especially in Android 7.0

```
Fatal Exception: java.lang.StackOverflowError: stack size 8MB
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:241)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:241)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:241)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:241)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:241)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:241)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:241)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:241)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:241)
       at com.bluelinelabs.conductor.ControllerHostedRouter.getTransactionIndexer(ControllerHostedRouter.java:241)
       at com.bluelinelabs.conductor.Router.ensureOrderedTransactionIndices(Router.java:910)
       at com.bluelinelabs.conductor.Router.getActivity(Router.java:412)
       at com.bluelinelabs.conductor.ControllerHostedRouter.setBackstack(ControllerHostedRouter.java:106)
       at com.bluelinelabs.conductor.Router.setRoot(Router.java:339)
       at classifieds.yalla.features.profile.my.ProfilePagerAdapter.configureRouter(ProfilePagerAdapter.kt:26)
       at com.bluelinelabs.conductor.support.RouterPagerAdapter.instantiateItem(RouterPagerAdapter.java:79)
       at android.support.v4.view.ViewPager.addNewItem(ViewPager.java:1004)
       at android.support.v4.view.ViewPager.populate(ViewPager.java:1152)
       at android.support.v4.view.ViewPager.populate(ViewPager.java:1086)
       at android.support.v4.view.ViewPager.onMeasure(ViewPager.java:1616)
       at android.view.View.measure(View.java:21125)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6461)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
       at android.view.View.measure(View.java:21125)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6461)
       at android.support.design.widget.CoordinatorLayout.onMeasureChild(CoordinatorLayout.java:739)
       at android.support.design.widget.HeaderScrollingViewBehavior.onMeasureChild(HeaderScrollingViewBehavior.java:91)
       at android.support.design.widget.AppBarLayout$ScrollingViewBehavior.onMeasureChild(AppBarLayout.java:1361)
       at android.support.design.widget.CoordinatorLayout.onMeasure(CoordinatorLayout.java:809)
       at android.view.View.measure(View.java:21125)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6461)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
       at android.view.View.measure(View.java:21125)
       at classifieds.yalla.features.main.TabMenuControllerLayout.onMeasure(TabMenuControllerLayout.kt:85)
       at android.view.View.measure(View.java:21125)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6461)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
       at android.view.View.measure(View.java:21125)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6461)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
       at android.support.v7.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:141)
       at android.view.View.measure(View.java:21125)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6461)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
       at android.view.View.measure(View.java:21125)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6461)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
       at android.view.View.measure(View.java:21125)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6461)
       at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1464)
       at android.widget.LinearLayout.measureVertical(LinearLayout.java:758)
       at android.widget.LinearLayout.onMeasure(LinearLayout.java:640)
       at android.view.View.measure(View.java:21125)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6461)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:185)
       at com.android.internal.policy.DecorView.onMeasure(DecorView.java:899)
       at android.view.View.measure(View.java:21125)
       at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:2612)
       at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1664)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1915)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1537)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7183)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:959)
       at android.view.Choreographer.doCallbacks(Choreographer.java:734)
       at android.view.Choreographer.doFrame(Choreographer.java:670)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:945)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6776)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1496)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1386)
```

**Comment by EricKuck** [comment]
I believe this was fixed a while ago in 926f7da. It's only in the snapshot build for now, but a 2.1.5 stable will be released soon.

**Comment by FunnyDevs** [comment]
I have encountered this kind of bug on 1 user on my crashlitycs app and i have 2.1.5 stable.
